underfs/oss/src/main/java/alluxio/underfs/oss/StsOssClientProvider.java

```diff
@@ -0,0 +1,239 @@
+    mRefreshOssClientScheduledThread.scheduleAtFixedRate(() -> {
+      try {
+        if (null != mOssConf) {
+          refreshOssStsClient(mOssConf);
+        }
+      } catch (Exception e) {
+        //retry it
+        LOG.warn("throw exception when clear meta data cache", e);
+      }
+    }, 0, 60000, TimeUnit.MILLISECONDS);
+  }
+
+  protected void refreshOssStsClient(UnderFileSystemConfiguration ossConfiguration)
+      throws IOException {
+    ClientBuilderConfiguration ossClientConf = getClientBuilderConfiguration(ossConfiguration);
+    createOrRefreshStsOssClient(ossConfiguration, ossClientConf);
+  }
+
+  private ClientBuilderConfiguration getClientBuilderConfiguration(
+      UnderFileSystemConfiguration ossConfiguration) {
+    ClientBuilderConfiguration ossClientConf = new ClientBuilderConfiguration();
+    ossClientConf.setMaxConnections(ossConfiguration.getInt(PropertyKey.UNDERFS_OSS_CONNECT_MAX));
+    ossClientConf.setMaxErrorRetry(ossConfiguration.getInt(PropertyKey.UNDERFS_OSS_RETRY_MAX));
+    ossClientConf.setConnectionTimeout(
+        (int) ossConfiguration.getMs(PropertyKey.UNDERFS_OSS_CONNECT_TIMEOUT));
+    ossClientConf.setSocketTimeout(
+        (int) ossConfiguration.getMs(PropertyKey.UNDERFS_OSS_SOCKET_TIMEOUT));
+    ossClientConf.setSupportCname(false);
+    ossClientConf.setCrcCheckEnabled(true);
+    return ossClientConf;
+  }
+
+  /**
+   * Init the STS OSS client.
+   * @param ossConfiguration OSS {@link UnderFileSystemConfiguration}
+   * @throws IOException if failed to init OSS client
+   */
+  public void initializeOssClient(UnderFileSystemConfiguration ossConfiguration)
+      throws IOException {
+    mOssConf = ossConfiguration;
+    if (null != mOssClient) {
+      return;
+    }
+
+    ClientBuilderConfiguration clientConf = getClientBuilderConfiguration(ossConfiguration);
+    createOrRefreshStsOssClient(ossConfiguration, clientConf);
+
+    LOG.info("init ossClient success : {}", mOssClient.toString());
+  }
+
+  boolean isStsTokenExpired() {
+    boolean expired = true;
+    Date now = convertLongToDate(System.currentTimeMillis());
+    if (null != mStsTokenExpiration) {
+      if (mStsTokenExpiration.after(now)) {
+        expired = false;
+      }
+    }
+    return expired;
+  }
+
+  boolean isTokenWillExpired() {
+    boolean in = true;
+    Date now = convertLongToDate(System.currentTimeMillis());
+    long millisecond = mStsTokenExpiration.getTime() - now.getTime();
+    if (millisecond >= mTokenTimeoutMs) {
+      in = false;
+    }
+    return in;
+  }
+
+  private void createOrRefreshStsOssClient(
+      UnderFileSystemConfiguration ossConfiguration,
+      ClientBuilderConfiguration clientConfiguration) throws IOException {
+    if (isStsTokenExpired() || isTokenWillExpired()) {
+      try {
+        String ecsRamRole = ossConfiguration.getString(PropertyKey.UNDERFS_OSS_ECS_RAM_ROLE);
+        String fullECSMetaDataServiceUrl = mEcsMetadataService + ecsRamRole;
```

Comment:
Need to check if it forms a correct URL if two strings are simply concatenated together. Also please leave a comment to the OSS docs where it explains how such a URL is formatted.

Reply:
mEcsMetadataService is a fixed URL, shouldn't be set by user